### PR TITLE
#11753 Fix sites end point issues

### PR DESCRIPTION
**Pull request by freddyucv**

### CODE REVIEW

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/site/SiteHelper.java [hunk 1]
```diff
@@ -201,4 +205,34 @@ public String getSelectedSite(final List<Host> siteList, final String siteInSess
...
+	 * Return the current site
+	 *
+	 * @param req
+	 * @return
+	 */
+	public Host getCurrentSite(final HttpServletRequest req, final User user) {
+		try {
+			final HttpSession session = req.getSession();
+			String hostId = (String) session.getAttribute(com.dotmarketing.util.WebKeys.CMS_SELECTED_HOST_ID);
+			Host host = null;
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v1/site/SiteHelper.java]: you can reduce it to

(null != hostId)?hostAPI.find(hostId, user, false):null;

File: dotCMS/src/main/java/com/dotcms/util/CollectionsUtils.java [hunk 2]
```diff
@@ -353,6 +353,16 @@
                 entry(key9, value9), entry(key10, value10));
     } // map.
 
+    public static <K,V> Map<K,V> map(final K key1, final V value1, final K key2, final V value2
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/util/CollectionsUtils.java]: A new one hehehe

File: dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/business/HostAPIImpl.java [hunk 3]
```diff
@@ -1010,31 +1010,53 @@ public void updateMenuLinks(Host workinghost,Host updatedhost) throws DotDataExc
...
-			Logger.error(HostAPIImpl.class, e.getMessage(), e);
-			throw new DotRuntimeException(e.getMessage(), e);
-		}
-	}
-	
+        try {
+            Structure st = CacheLocator.getContentTypeCache().getStructureByVelocityVarName("Host");
+            String condition="";
+
+            if(showArchived){
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/business/HostAPIImpl.java]: you can reduce it to condition = (showArchived)?" +deleted:true":" +deleted:false";

Comment (jdotcms) [inline dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/business/HostAPIImpl.java]: Condition could be a StringBuilder

File: dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/business/HostAPI.java [hunk 4]
```diff
@@ -307,4 +307,14 @@
...
 	public PaginatedArrayList<Host> search(String filter, boolean showArchived, boolean showSystemHost, int limit, int offset, User user, boolean respectFrontendRoles) throws DotDataException, DotSecurityException;
 
+	/**
+	 * Return the number of sites for user
+	 *
+	 * @param user
+	 * @param respectFrontendRoles
+	 * @return
+	 * @throws DotDataException
+	 * @throws DotSecurityException
```
Comment (dsilvam) [inline dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/business/HostAPI.java]: {DotDataException, DotSecurityException } are not really thrown

File: dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/business/HostAPIImpl.java [hunk 5]
```diff
@@ -1010,31 +1010,53 @@ public void updateMenuLinks(Host workinghost,Host updatedhost) throws DotDataExc
...
+    }
+
+    /**
+     * Return the number of sites for user
+     *
+     * @param user
+     * @param respectFrontendRoles
+     * @return
+     * @throws DotDataException
+     * @throws DotSecurityException
```
Comment (dsilvam) [inline dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/business/HostAPIImpl.java]: {DotDataException, DotSecurityException } are not really thrown

**Review by dsilvam — CHANGES_REQUESTED** [review]

**Review by dsilvam — APPROVED** [review]

**Review by jdotcms — APPROVED** [review]

**Review by jgambarios — APPROVED** [review]

**Pull request merged**